test/e2e/cluster.go

```diff
@@ -83,6 +83,19 @@ var _ = Describe("Cluster smoke test", func() {
 			return ready.ServiceIsReady(svc)
 		}, 5*time.Minute, 10*time.Second).Should(BeTrue())
 	})
+
+	// mainly we want to test the gateway/egress functionality - this request for the image will travel from
+	// node > gateway > storage account of the registry.
+	Specify("Can access and use the internal container registry", func() {
+		ctx := context.Background()
+		deployName := "internal-registry-deploy"
+		err := createContainerFromInternalContainerRegistryImage(ctx, clients.Kubernetes, deployName)
+		Expect(err).NotTo(HaveOccurred())
+
+		err = wait.PollImmediate(10*time.Second, 15*time.Minute, ready.CheckDeploymentIsReady(ctx, clients.Kubernetes.AppsV1().Deployments(testNamespace), deployName))
```

Comment:
I'm tempted to drop the overarching poll time on this.  It shouldn't take 15 minutes to have this deployment run + pull the image.  

I know in other situations it is less, but I think it should be ~5 minutes instead of 15 because we're not doing a lot here.

Reply:
I was thinking about this too - OK.